test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go

```diff
@@ -217,6 +221,25 @@ var _ = SIGDescribe("[HPA] Horizontal pod autoscaling (scale resource: Custom Me
 			hpa:             externalHPA(f.Namespace.ObjectMeta.Name, metricTargets)}
 		tc.Run()
 	})
+
+	ginkgo.It("should scale down to 0 with Custom Metric of type Object from Stackdriver [Feature:CustomMetricsAutoscaling]", func() {
+		defer withFeatureGate(features.HPAScaleToZero, true)()
```

Comment:
I don't really understand what this is doing... what effect does changing a feature gate in the e2e binary have? it doesn't change any behavior in the system under test

Reply:
This was removed as a part of https://github.com/kubernetes/kubernetes/pull/112444.